athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java

```diff
@@ -140,36 +139,24 @@ protected void readWithConstraint(final BlockSpiller spiller, final ReadRecordsR
 
             GraphTraversal<Vertex, Map<Object, Object>> graphTraversalFinal = graphTraversal.valueMap();
 
-            //log string equivalend of gremlin query
-            logger.info("readWithConstraint: enter - " + 
-             GroovyTranslator.of("g").translate(graphTraversalFinal.asAdmin().getBytecode()));
+            // log string equivalent of gremlin query
+            logger.info("readWithConstraint: enter - "
+                    + GroovyTranslator.of("g").translate(graphTraversalFinal.asAdmin().getBytecode()));
+
+            GeneratedRowWriter.RowWriterBuilder builder = GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+
+            for (final Field nextField : recordsRequest.getSchema().getFields()) {
+                builder = TypeRowWriter.writeRowTemplate(builder, nextField);
```

Comment:
Why are we returning builder here? seems unnecessary given that you are mutating the input builder. The return has no affect.

Reply:
re-factored